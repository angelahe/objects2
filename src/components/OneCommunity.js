import React from 'react';
import '../styles/Styles140.css';
import editbtn from '../images/edit_FFFFFF.png';
import deletebtn from '../images/delete_FFFFFF.png';

const OneCommunity = (props) => {

  function handleAccountClick(e) {
    switch(e.target.getAttribute("elemtype")) {
      case "Edit":
        props.editClicked(props.city); break;
      case "Delete":
        props.deleteClicked(props.city); break;
      case "Container":
        props.cityClicked(props.city); break;
      case "Text":
        props.cityClicked(props.city); break;
      default:
        console.log("unknown element clicked for city");
    }
  }

  const communityType = props.city.howBig();

  return (
    <div className="ItemBox AppListItem" elemtype = "Container" onClick={handleAccountClick}>
        <h2>{communityType} {" of "} {props.city.Name}</h2>
        <span className="DetailText" elemtype="Text">Latitude: {props.city.Latitude}</span><br/>
        <span className="DetailText" elemtype="Text">Longitude: {props.city.Longitude}</span><br/>
        <span className="DetailText" elemtype="Text">Population: {props.city.Population}</span><br/>

        <button className="AppBtn" elemtype = "Edit">
          <img className="btnImg" elemtype = "Edit" src={editbtn} alt="Add"/>
        </button>
        <button className="AppBtn Delete" elemtype = "Delete">
          <img className="btnImg" elemtype = "Delete" src={deletebtn} alt="Add"/>
        </button>
        <hr/>
    </div>
  )
}

export default OneCommunity;

