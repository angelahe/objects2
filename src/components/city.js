class city {

  constructor(name, latitude, longitude, population) {
    this.Name = name;
    this.Latitude = latitude;
    this.Longitude = longitude;
    this.Population = population;
  }

  show() {
    const attributeString = `${this.Name} ${this.Latitude} ${this.Longitude} ${this.Population}`;
    return attributeString;
  }

  movedIn(numPeople) {
    const populationIn = numPeople;
    return this.Population += populationIn;
  }

  movedOut(numPeople) {
    const populationOut = numPeople;
    return this.Population -= populationOut;
  }

  howBig() {
    switch (true) {
      case (this.Population > 100000) :
        return "City";
        break;
      case (this.Population > 20000) :
        return "Large town";
        break;
      case (this.Population >1000) :
        return "Town";
        break;
      case (this.Population > 100) :
        return "Village";
        break;
      case(this.Population > 0):
        return "Hamlet";
        break;
        default:
          return "Not sure how big this place is...";
    }
  }



}

export default city