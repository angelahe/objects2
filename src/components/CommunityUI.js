import React from 'react';
import community from './community';
import CityCreate from './CityCreate';
import CityEdit from './CityEdit';
import CityDelete from './CityDelete';
import OneCommunity from './OneCommunity';
import CommunitySummary from './CommunitySummary';
import CityActions from './CityActions';
import addbtn from '../images/add_FFFFFF.png';

import '../styles/Styles140.css';

class CommunityUI extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      CommunityList: new community(),
      currentCity: null,
      addShow: false,
      editShow: false,
      actionsShow: false,
      deleteShow: false
    }
  }

  onBtnCloseClick = (closeType) => {
    if (closeType === "create") this.setState ({addShow: false});
    if (closeType === "edit") this.setState ({editShow: false});
    if (closeType === "delete") this.setState ({deleteShow: false});
    if (closeType === "actions") this.setState ({actionsShow: false});
  }

  onBtnAddClick = () => {
    this.setState({addShow: true});
  }

  onBtnEditClick = () => {
    this.setState({editShow: true});
  }

  onCityClick = (city) => {
    this.setState({
      actionsShow: true,
      currentCity: city
    })
  }

  onEditClick = (city) => {
    this.setState({
      editShow: true,
      currentCity: city
    })
  }

  onDeleteClick = (city) => {
    this.setState({
      deleteShow: true,
      currentCity: city
    })
  }

  onCityDelete = (city) => {
    this.state.CommunityList.deleteCity(city);

    this.setState({
      CommunityList: this.state.CommunityList,
      deleteShow: false
    })
  }

  onCityUpdate = (city, updated) => {
    console.log("in city update");
    this.state.CommunityList.updateCommunity(city, updated);

    this.setState({
      CommunityList: this.state.CommunityList,
      editShow: false
    })
  }


  onCityCreate = (name, latitude, longitude, population) => {
    this.state.CommunityList.addCommunity(name, latitude, longitude, population);

    this.setState({
      CommunityList: this.state.CommunityList,
      addShow: false
      }
    )

  }

  onMovedIn = (city, number) => {
    console.log("in moved in number is", number);
    this.state.CommunityList.movedIn(city, number);

    this.setState({
      CommunityList: this.state.CommunityList,
      actionsShow: false,
    });
  }

  onMovedOut = (city, number) => {
    console.log("in moved out number is", number);
    this.state.CommunityList.movedOut(city, number);

    this.setState({
      CommunityList: this.state.CommunityList,
      actionsShow: false,
    });
  }

  render() {
    const communities = this.state.CommunityList.Communities.map((city, index) =>
    <OneCommunity key = {city.Name}
                  city = {city}
                  cityClicked={this.onCityClick}
                  editClicked={this.onEditClick}
                  deleteClicked={this.onDeleteClick}
    />
    );

    return(
      <div>

        <h1>This is the Community UI</h1>
        <div className="AppContainer">
          <CommunitySummary communities={this.state.CommunityList}/>
          <div className="AppPanel">
            <div className="ItemBox AppHeader">
              <span className="AddItem">Add City</span>
              <button className="AppBtn" onClick={this.onBtnAddClick}>
                <img className="btnImg" src={addbtn} alt="Add"/>
              </button>
            </div>
            <div className = "AppList">
              {communities}
            </div>
          </div>
          <div className = "AppPanel">
            {(this.state.addShow)
              ? <CityCreate createClicked = {this.onCityCreate}
                            closeClicked={this.onBtnCloseClick}
                />
              :null
            }
            {(this.state.editShow)
              ? <CityEdit
                         city={this.state.currentCity}
                         updateCity = {this.onCityUpdate}
                         closeClicked={this.onBtnCloseClick}
                />
              : null
              }
            {(this.state.deleteShow)
              ? <CityDelete
                        city={this.state.currentCity}
                        deleteCity={this.onCityDelete}
                        closeClicked={this.onBtnCloseClick}
                />
              :null
              }
            {(this.state.actionsShow)
              ? <CityActions
                city={this.state.currentCity}
                movedIn={this.onMovedIn}
                movedOut={this.onMovedOut}
              />
              : null
            }
          </div>

        </div>
      </div>
    )
  }
}
export default CommunityUI
