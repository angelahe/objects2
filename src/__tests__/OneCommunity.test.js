import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import OneCommunity from "../components/OneCommunity";
import CommunityUI from "../components/CommunityUI";
import city from "../components/city";
import renderer from "react-test-renderer";

const props = new CommunityUI();
const thiscity = new city("Calgary", "100 N", "100 W", 1000000);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<OneCommunity key = {thiscity.Name}
                                city = {thiscity}
                                cityClicked={props.onCityClick}
                                editClicked={props.onEditClick}
                                deleteClicked={props.onDeleteClick}
  />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  shallow(<OneCommunity key = {thiscity.Name}
                      city = {thiscity}
                      cityClicked={props.onCityClick}
                      editClicked={props.onEditClick}
                      deleteClicked={props.onDeleteClick}
  />);
});

test('Starter renders correctly', () => {
  const tree = renderer.create(<OneCommunity key = {thiscity.Name}
                                           city = {thiscity}
                                           cityClicked={props.onCityClick}
                                           editClicked={props.onEditClick}
                                           deleteClicked={props.onDeleteClick}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});