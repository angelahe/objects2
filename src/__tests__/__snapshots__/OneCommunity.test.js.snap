// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Starter renders correctly 1`] = `
<div
  className="ItemBox AppListItem"
  elemtype="Container"
  onClick={[Function]}
>
  <h2>
    City
     
     of 
     
    Calgary
  </h2>
  <span
    className="DetailText"
    elemtype="Text"
  >
    Latitude: 
    100 N
  </span>
  <br />
  <span
    className="DetailText"
    elemtype="Text"
  >
    Longitude: 
    100 W
  </span>
  <br />
  <span
    className="DetailText"
    elemtype="Text"
  >
    Population: 
    1000000
  </span>
  <br />
  <button
    className="AppBtn"
    elemtype="Edit"
  >
    <img
      alt="Add"
      className="btnImg"
      elemtype="Edit"
      src="edit_FFFFFF.png"
    />
  </button>
  <button
    className="AppBtn Delete"
    elemtype="Delete"
  >
    <img
      alt="Add"
      className="btnImg"
      elemtype="Delete"
      src="delete_FFFFFF.png"
    />
  </button>
  <hr />
</div>
`;
